umbra/umbra_32.c

```diff
@@ -1049,3 +1049,42 @@ umbra_handle_fault(void *drcontext, byte *target, dr_mcontext_t *raw_mc,
 {
     return false;
 }
+
+drmf_status_t
+umbra_clear_redundant_blocks(umbra_map_t *map)
+{
+    byte *shadow_data;
+    uint i, j;
+
+    if (map == NULL)
+        return DRMF_ERROR_INVALID_PARAMETER;
+
+    /* Umbra needs to have create-on-touch optimization enabled. */
+    if (!TEST(UMBRA_MAP_CREATE_SHADOW_ON_TOUCH, map->options.flags))
+        return DRMF_ERROR;
```

Comment:
Is there a more specific error that would fit: DRMF_ERROR_FEATURE_NOT_AVAILABLE or DRMF_ERROR_INVALID_PARAMETER?

Reply:
Done.